src/cluster_legacy.c

```diff
@@ -4928,6 +4928,12 @@ int clusterDelSlot(int slot) {
 
     if (!n) return C_ERR;
 
+    /* Cleanup the empty dicts if myself is the owner. */
+    if (n == myself) {
```

Comment:
This isn't great since since replicas will keep this populated, since n != myself in that case. You could also add the check similar to what we do in `getMyShardSlotCount`. I vaguely remember a command for, "doesMyShardOwnSlot()", but it's probably in a pending CR that I can't immediately find.